File: Campus_shops/src/main/java/com/entity/Recommend.java
package com.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.Date;

/**
 * <p>
 * 推荐商品实体类。对应数据表recommend
 * </p>
 * 属性和字段一致
 */
@AllArgsConstructor//全参构造
@NoArgsConstructor//无参构造
@Data
@Accessors(chain = true)//链式写法
public class Recommend {
    private static final long serialVersionUID = 1L;

    /**
     *主键
     */
    private String id;
    /**
     * 商品id，商品的唯一标识。commodity表的主键
     */
    private String commid;
    /**
     * 显示在主页的商品名称。不一定是商品名称（后者有时过于冗长）
     */
    private String recomname;
    /**
     * 显示在主页的商品图片
     * 需要重新上传，不一定是商品主图（为了营销会做新图），但必须png且背景透明
     */
    private String recomimg;
    /**
     * 显示在主页的推荐语录
     * 不一定是商品描述
     */
    private String recomdesc;
    /**
     * 推荐商品的状态
     *
     * 本来：0待审核、1正在首页展示、2审核通过待首页展示、3审核不通过、4商品已被下架
     * 0表示待审核，1表示审核通过并显示在首页（即正在首页推荐展示），2表示审核通过但是待推荐（最多推荐2个商品），即暂时还不显示在首页推荐
     * 3表示审核不通过，4表示商品已经下架，不再显示在首页（与commodity的状态4一致）
     *
     * 现在变更为：
     * 0表示待审核，
     * 1表示审核通过（会从中按时间顺序选两个展示在首页）
     * 2表示审核不通过，
     * 3表示商品已经下架（与commodity的状态4一致）
     */
    private Integer recomstatus;
    /**
     * 更新时间（用户请求推荐、审核通过等）
     */
    private Date updatetime;
}

File: Campus_shops/src/main/resources/static/js/admin/recommend.js
layui.use(['form', 'element', 'util', 'carousel', 'laypage', 'layer', 'table'], function () {
    var element = layui.element;
    var util = layui.util;
    var form = layui.form;
    var laypage = layui.laypage
        , layer = layui.layer;
    form.on('select(types)', function (data) {
        var indexGID = data.elem.selectedIndex;
        lookallrecommend(data.elem[indexGID].title);
    });
});
function lookallrecommend(recomstatus) {
    layui.use(['form', 'element', 'util', 'carousel', 'laypage', 'layer','table'], function () {
        var table = layui.table;
        table.render({
            elem: '#recommend'
            , url: basePath+'/admin/recommend/'+recomstatus
            , page: {
                layout: ['limit', 'count', 'prev', 'page', 'next', 'skip']
                , groups: 3
                , limits: [20, 50, 100]
                , limit: 20
            }, cols: [[
                {field: 'qid', title: 'ID',width:80, align:'center'}
                , {field: 'commid', title: '商品id', width: 150, align:'center'}
                , {field: 'recomname', title: '推荐名称', width: 157, align:'center'}
                , {field: 'recomimg', title: '推荐图片', width: 150, align:'center'}
                , {field: 'recomdesc', title: '推荐描述', width: 200, align:'center'}
                , {field: 'updatetime', title: '时间', width: 200,sort: true, align:'center'}
                , {field: 'recomstatus', title: '商品推荐状态',  toolbar: '#recommendStatusDemo', width: 150, align:'center'}
                , {fixed: 'right', title: '操作', toolbar: '#barDemo', width:200, align:'center'}
            ]], done: function (res, curr, count) {
                var i=1;
                $("[data-field='qid']").children().each(function () {
                    if($(this).text() == 'ID') {
                        $(this).text("ID")
                    }else{
                        $(this).text(i++)
                    }
                });
            }
        });
        //监听行工具事件
        table.on('tool(test)', function (obj) {
            var data = obj.data;
            if (obj.event === 'xiangqing') {
                window.open(basePath+"/product-detail/"+data.commid)
            }else if(obj.event === 'weigui'){
                layer.confirm('确认审核不通过吗？', {
                    btn: ['确定','算了'], //按钮
                    title:"推荐商品审核不通过",
                    offset:"50px"
                }, function(){
                    layer.closeAll();
                    $.ajax({
                        url: basePath+'/admin/changerecomstatus/'+data.commid+"/2",
                        data: "",
                        contentType: "application/json;charset=UTF-8", //发送数据的格式
                        type: "put",
                        dataType: "json", //回调
                        beforeSend: function () {
                            layer.load(1, { //icon支持传入0-2
                                content: '请稍等...',
                                success: function (layero) {
                                    layero.find('.layui-layer-content').css({
                                        'padding-top': '39px',
                                        'width': '60px'
                                    });
                                }
                            });
                        },
                        complete: function () {
                            layer.closeAll('loading');
                        },
                        success: function (data) {
                            console.log(data)
                            if(data.status===200){
                                layer.msg(data.message, {
                                    time: 1000,
                                    icon: 1,
                                    offset: '50px'
                                }, function () {
                                    location.reload();
                                });
                            }else {
                                layer.msg(data.message, {
                                    time: 1000,
                                    icon: 2,
                                    offset: '50px'
                                });
                            }
                        }
                    });
                }, function(){
                });
            }else if (obj.event === 'shenhe') {
                layer.confirm('确认审核通过吗？', {
                    btn: ['确定','算了'], //按钮
                    title:"推荐商品审核通过",
                    offset:"50px"
                }, function(){
                    layer.closeAll();
                    $.ajax({
                        url: basePath+'/admin/changerecomstatus/'+data.commid+"/1",
                        data: "",
                        contentType: "application/json;charset=UTF-8", //发送数据的格式
                        type: "put",
                        dataType: "json", //回调
                        beforeSend: function () {
                            layer.load(1, { //icon支持传入0-2
                                content: '请稍等...',
                                success: function (layero) {
                                    layero.find('.layui-layer-content').css({
                                        'padding-top': '39px',
                                        'width': '60px'
                                    });
                                }
                            });
                        },
                        complete: function () {
                            layer.closeAll('loading');
                        },
                        success: function (data) {
                            console.log(data)
                            if(data.status===200){
                                layer.msg(data.message, {
                                    time: 1000,
                                    icon: 1,
                                    // offset: '50px'
                                }, function () {
                                    location.reload();
                                });
                            }else {
                                layer.msg(data.message, {
                                    time: 1000,
                                    icon: 2,
                                    // offset: '50px'
                                });
                            }
                        }
                    });
                }, function(){
                });
            }
        });
    });
}
lookallrecommend(100);